src/engines/html5/html5.js

```diff
@@ -1145,6 +1148,18 @@ export default class Html5 extends FakeEventTarget implements IEngine {
     }
   }
 
+  _handleWaiting(): void {
+    let playing = false;
+    this._eventManager.listenOnce(this._el, Html5EventType.PLAYING, () => (playing = true));
+    setTimeout(() => {
+      // prevent sending waiting event for too short buffering
+      if (!playing) {
+        this.dispatchEvent(new FakeEvent(Html5EventType.WAITING));
```

Comment:
so if it's not playing WAITING events will keep being fired?

Reply:
as it should

Reply:
do we need to fire the WAITING manually ? doesnt it fire anyway if the video is waiting ?

Reply:
yes we do. the HTML5 class warps the video element and delegates its event outside see https://github.com/kaltura/playkit-js/blob/80afd171528293b1a310f0ac5cc1a5b216dfa5fa/src/engines/html5/html5.js#L302